pkg/kedge/grpc/director/adhoc/adhoc.go

```diff
@@ -0,0 +1,50 @@
+package adhoc
+
+import (
+	"net"
+	"strconv"
+
+	"github.com/improbable-eng/kedge/pkg/kedge/common"
+	"github.com/improbable-eng/kedge/pkg/kedge/grpc/director/router"
+	"github.com/improbable-eng/kedge/protogen/kedge/config/common"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+type static struct {
+	rules []*kedge_config_common.Adhoc
+}
+
+func NewStaticAddresser(rules []*kedge_config_common.Adhoc) *static {
+	return &static{rules: rules}
+}
+
+func (a *static) Address(hostString string) (string, error) {
+	hostName, port, err := common.ExtractHostPort(hostString)
+	if err != nil {
+		return "", grpc.Errorf(codes.InvalidArgument, "adhoc: malformed port number: %v", err)
```

Comment:
Let's maybe we start using non deprecated package for gRPC errors? at least for new code? (: the new one is `status.Errorf`

Here and below